ft-email-auth-provider/src/handlers/create_account.rs

```diff
@@ -321,3 +177,199 @@ pub fn create_account(
 
     Ok(ft_sdk::form::redirect("/")?.with_cookie(auth::session_cookie(sid.as_str(), host)?))
 }
+
+#[derive(serde::Deserialize)]
+struct CreateAccountPayload {
+    email: String,
+    #[cfg(feature = "username")]
+    username: String,
+    name: String,
+    password: String,
+    password2: String,
+    accept_terms: bool,
+}
+
+impl CreateAccountPayload {
+    pub(crate) fn validate(
+        &self,
+        conn: &mut ft_sdk::Connection,
+        errors: &mut std::collections::HashMap<String, String>,
+    ) -> Result<(), ft_sdk::Error> {
+        if !CreateAccountPayload::validate_email(&self.email) {
+            errors.insert("email".to_string(), "invalid email format".to_string());
+        }
```

Comment:
```suggestion
        if !self.validate_email() {
            errors.insert("email".to_string(), "invalid email format".to_string());
        }
```

Reply:
Also, let's use the right error message. Since this is shown to the user. Always capitalise the first letter and use full stop(.) at the end.